packages/core/src/events/inference/retrieve/text/remote/base.ts

```diff
@@ -0,0 +1,86 @@
+import { SecretFields } from '../../../../../load/keymap.js';
+import {
+  AsyncCaller,
+  AsyncCallerParams,
+} from '../../../../../utils/asyncCaller.js';
+import { Context } from '../../../../input/load/docs/context.js';
+import { BaseTextRetriever, BaseTextRetrieverParams } from '../../base.js';
+import { RemoteRetrieverAuth, RemoteRetrieverValues } from './index.js';
+
+export interface RemoteRetrieverParams
+  extends AsyncCallerParams,
+    BaseTextRetrieverParams {
+  url: string;
+
+  auth: RemoteRetrieverAuth;
+}
+
+export abstract class RemoteRetriever
```

Comment:
What is this class for?

Reply:
Like local nodes? Why does the code seems like we are making a POST request somewhere though?